Mono.Debugging.Soft/SoftDebuggerSession.cs

```diff
@@ -737,19 +739,84 @@ protected override ProcessInfo[] OnGetProcesses ()
 		internal PortablePdbData GetPdbData (AssemblyMirror asm)
 		{
 			string assemblyFileName;
+
 			if (!assemblyPathMap.TryGetValue (asm.GetName ().FullName, out assemblyFileName))
 				assemblyFileName = asm.Location;
 			var pdbFileName = Path.ChangeExtension (assemblyFileName, ".pdb");
-			if (!PortablePdbData.IsPortablePdb (pdbFileName))
-				return null;
-			return new PortablePdbData (pdbFileName);
+			if (PortablePdbData.IsPortablePdb (pdbFileName))
+				return new PortablePdbData (pdbFileName);
+			// Attempt to fetch pdb from the debuggee over the wire
+			var pdbBlob = asm.GetPdbBlob ();
+			return pdbBlob != null ? new PortablePdbData (pdbBlob) : null;
 		}
 
 		internal PortablePdbData GetPdbData (MethodMirror method)
 		{
 			return GetPdbData (method.DeclaringType.Assembly);
 		}
 
+		Dictionary<AssemblyMirror, SourceLinkMap []> SourceLinkCache = new Dictionary<AssemblyMirror, SourceLinkMap []> ();
+		SourceLinkMap [] GetSourceLinkMaps (MethodMirror method)
+		{
+			var asm = method.DeclaringType.Assembly;
+			SourceLinkMap [] maps = Array.Empty<SourceLinkMap>();
+
+			if (SourceLinkCache.TryGetValue (asm, out maps)) {
+				return maps;
+			}
+
+			string jsonString = null;
```

Comment:
The code assumed the maps array would remain non-null after TryGetValue is done, but it would override it on failure. In case we would try to add a `null` to the map below.

```suggestion
			maps = Array.Empty<SourceLinkMap> ();
			string jsonString = null;
```

Reply:
Fix for this: https://github.com/mono/debugger-libs/pull/238/files#r310340014

Reply:
I'm not understanding this suggestion

Reply:
Tl;dr:

Variable is set to Array.Empty.
Dictionary.TryGetValue will reset it to null.
If no mapping is found, it'll remain nu, thus NRE

Reply:
Due to now doing `maps ?? Array.Empty` the bug is no longer there.

I thought we weren't null checking return values, so it would've ended up in NRE.